IzvorniKod/Views/Home/Roles.cshtml

```diff
@@ -4,60 +4,76 @@
-@if (User.IsInRole("Reviewer"))
-{
-    <p>Reviewer</p>
-}
-@if (User.IsInRole("Educator"))
-{
-    <p>Educator</p>
-}
-@if (User.IsInRole("SuperStudent"))
+            <p>Get a <strong>Student</strong> role by clicking the button below.</p>
+
+            <form class="text-start mb-3" asp-action="AddUserToSuperStudentRole" asp-controller="Home" method="post">
+                <input type="hidden" name="roleName" value="SuperStudent" />
+                <button type="submit" class="btn btn-sm btn-primary">Get Student Role</button>
+            </form>
+
+            <form asp-action="AddUserRoleStatus" asp-controller="Home" method="post" class="apply-role">
+                <input type="hidden" name="roleName" value="Educator" />
+                <div class="form-group">
+                    <label for="description" class="fw-bold">Educator</label>
+                    <p>If you're applying for the Educator role, please enter any relevant information in the field below.</p>
+                    <textarea id="textarea1" name="description" class="form-control resize-disable form-bg-color" placeholder="Write a bit about yourself..." rows="5"></textarea>
+                </div>
+                <div class="button-role mt-lg-4 mt-5 pt-lg-0 pt-5">
+                    <button id="EducatorReset" type="button" class="btn btn-primary" disabled>Reset</button>
+                    <button type="submit" class="btn btn-primary">Send</button>
+                </div>
+            </form>
+
+            <form asp-action="AddUserRoleStatus" asp-controller="Home" method="post" class="apply-role mt-4">
+                <input type="hidden" name="roleName" value="Reviewer" />
+                <div class="form-group">
+                    <label for="description" class="fw-bold">Reviewer</label>
+                    <p>If you're applying for the Reviewer role, please enter any relevant information in the field below.</p>
+                    <textarea id="textarea2" name="description" class="form-control resize-disable form-bg-color" placeholder="Write a bit about yourself..." rows="5"></textarea>
+                </div>
+                <div class="button-role mt-lg-4 mt-5 pt-lg-0 pt-5">
+                    <button id="ReviewerReset" type="button" class="btn btn-primary" disabled>Reset</button>
+                    <button type="submit" class="btn btn-primary">Send</button>
+                </div>
+            </form>
+
+            @if (appliedForRole == true)
+            {
+                <script>
+                    alert('Successfully applied for role!');
+                </script>
+            }
+            @if (userRoleStatus == "Unknown")
+            {
+                <p class="text-center fw-bold mt-1">Appliement is either rejected or you haven't yet applied for the role.</p>
+            }
+            @if (userRoleStatus != "Unknown")
+            {
+                <p class="text-center fw-bold mt-1">Status: @userRoleStatus</p>
```

Comment:
Tijekom testiranja sam primjetio da neki user koji se već prijavio na neki role ili mu je on approvean ne može nikako znati da ne može poslati još jedan application. Možda da se napravi jednostavni javascript kod koji provjerava sadržaj p elementa u kojem piše Pending i Approved i onda, ako je trenutačno tu upisana jedna od te dvije riječi, da mu se ispiše da ne može slati još applicationa i da mu se disableaju send buttoni. Ako postoji neki lakši način za to, može i tako naravno.

Reply:
Stavio sam da su form i buttons (Send button) disabled ukoliko je userRoleStatus različit od "Unknown", dakle ili Approved ili Pending. Ako je role dodijeljen korisniku preko baze (npr. kako je Adminu), njemu neće biti disabled, ali se neće niti moći prijaviti.